framework/src/dgkernels/DGDiffusionInterface.C

```diff
@@ -0,0 +1,90 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "DGDiffusionInterface.h"
+
+#include <cmath>
+
+template<>
+InputParameters validParams<DGDiffusionInterface>()
+{
+  InputParameters params = validParams<DGInterface>();
+  params.addParam<Real>("D",1.,"The diffusion coefficient.");
+  params.addParam<Real>("D_neighbor",1.,"The neighboring diffusion coefficient.");
+  return params;
+}
+
+DGDiffusionInterface::DGDiffusionInterface(const InputParameters & parameters) :
+    DGInterface(parameters),
+    _D(getParam<Real>("D")),
+    _D_neighbor(getParam<Real>("D_neighbor"))
+{
+  if (!parameters.isParamValid("boundary"))
+  {
+    mooseError("In order to use the DGDiffusionInterface dgkernel, you must specify a boundary where it will live.");
+  }
+}
+
+Real
+DGDiffusionInterface::computeQpResidual(Moose::DGResidualType type)
+{
+  Real r = 0;
+
+  switch (type)
```

Comment:
The difference in the two cases seems to be a minus sign and which test function you multiply by.  So I'd initialize `r` with:
```
r = 0.5 * (-_D * _grad_u[_qp] * _normals[_qp] + -_D_neighbor * _grad_neighbor_value[_qp] * _normals[_qp]);
```

then return either `r * _test[_i][_qp]` or `-r * _test_neighbor[_i][_qp]` directly from the cases.